src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -10792,6 +10791,16 @@ and TcNormalizedBinding declKind (cenv: cenv) env tpenv overallTy safeThisValOpt
                 errorR(Error(FSComp.SR.tcLiteralCannotBeInline(), mBinding))
             if not (isNil declaredTypars) then
                 errorR(Error(FSComp.SR.tcLiteralCannotHaveGenericParameters(), mBinding))
+            
+        if g.langVersion.SupportsFeature(LanguageFeature.EnforceAttributeTargetsOnFunctions) && memberFlagsOpt.IsNone && not attrs.IsEmpty then
```

Comment:
Note: this PR only covers `module` and `class` let bindings for now. Will create follow up PR for `members, properties` etc

Reply:
Cool!